frontend/src/contexts/apolloClientContext.tsx

```diff
@@ -8,33 +8,49 @@ import { setContext } from "@apollo/client/link/context";
 import { ReactNode, useContext, useEffect, useState } from "react";
 import { User, UserContext } from "../contexts/userContext";
 import { Roles } from "../utils/types";
+import { auth } from "../../firebaseConfig"; // Ensure this is the path to your Firebase configuration
 
 export const GRAPHQL_URI = "http://127.0.0.1:9191/v1/graphql";
 
 const httpLink = createHttpLink({
   uri: GRAPHQL_URI,
 });
 
-const createAuthLink = (user?: User) =>
-  setContext((_, { headers }) => {
+const createAuthLink = (user?: User, tokenBypass?: string) =>
+  setContext(async (_, { headers }) => {
     const roleHeader =
       user && user.userId !== "unauthenticated"
         ? {
             "x-hasura-user-id": user.userId,
             "x-hasura-role": user.role.toLowerCase(),
           }
         : { "x-hasura-role": Roles.UNAUTHENTICATED_USER };
+
+    let token = "";
```

Comment:
It will be safer to use undefined
```suggestion
    let token: string | undefined;
```